components/js/override.js

```diff
@@ -22,6 +22,14 @@ define(['angular'], function(angular) {
     .constant('OVERRIDE', {
       'APP_OPTIONS': {
         'optionsTemplateURL': 'portal/misc/partials/example-options.html',
+        'appMenuTemplateURL': 'portal/misc/partials/example-menu.html',
```

Comment:
What if the API were a URL that returned JSON representing the menu? Once it's on the other end of a URL, it can be arbitrarily implemented, whether as a static file or as a dynamic back end that considers user identity.

I think I'd prefer to define a semantic JSON API for menu and have the `uportal-app-framework` render that rather than escape to an arbitrary template HTML, so that we can semantically version the API. We can then engineer the framework to continue to support the JSON across versions of the framework, regardless of the evolution of the implementation of the menu.

Reply:
I agree with you. This quick and dirty option is meant to meet two of those requirements that have already come up in conversation:

- Show/hide items depending upon user attributes or group memberships
- Menu items that call JavaScript rather than just hyperlink a URL

I'm having a hard time envisioning how all those pieces would come together, but I would like to establish a better way of doing this.